src/VisualStudio/Core/Def/Implementation/SymbolSearch/SymbolSearchResult.cs

```diff
@@ -0,0 +1,232 @@
+                }
+            }
+            else if (definitionItem != null)
+            {
+                result.IsDefinition = true;
+                if (definitionItem.DisplayableProperties.TryGetValue(AbstractReferenceFinder.ContainingTypeInfoPropertyName, out var containingTypeInfo))
+                {
+                    result.ContainingTypeName = containingTypeInfo;
+                }
+
+                if (definitionItem.DisplayableProperties.TryGetValue(AbstractReferenceFinder.ContainingMemberInfoPropertyName, out var containingMemberInfo))
+                {
+                    result.ContainingMemberName = containingMemberInfo;
+                }
+                result.Kind = "Definition";
+
+                result.Icon = new ImageElement(definitionItem.Tags.GetFirstGlyph().GetImageId());
+
+                // TODO: Use DocumentSpanEntry
+                result.ContextText = new ClassifiedTextElement(excerptResult.ClassifiedSpans.Select(cspan =>
+                    new ClassifiedTextRun(cspan.ClassificationType, sourceText.ToString(cspan.TextSpan))));
+
+                result.ContextHighlightSpan = AsSpan(excerptResult.MappedSpan);
+                // TODO: add definition context and its highlight
+            }
+
+            // Set location so that Editor can navigate to the result
+            var mappedDocumentSpan = await FindUsagesUtilities.TryMapAndGetFirstAsync(documentSpan, sourceText, cancellationToken).ConfigureAwait(false);
+            if (mappedDocumentSpan.HasValue)
+            {
+                var location = mappedDocumentSpan.Value;
+
+                result.RelativePath = documentSpan.Document.FilePath;
+                result.LineNumber = location.LinePositionSpan.Start.Line;
+                result.CharacterNumber = location.LinePositionSpan.Start.Character;
+
+                result.PersistentSpan = context.SymbolSource.ServiceProvider.PersistentSpanFactory.Create(
+                    documentSpan.Document.FilePath,
+                    location.LinePositionSpan.Start.Line,
+                    location.LinePositionSpan.Start.Character,
+                    location.LinePositionSpan.End.Line,
+                    location.LinePositionSpan.End.Character,
+                    SpanTrackingMode.EdgeInclusive);
+            }
+
+            return result;
+        }
+
+        private static Span AsSpan(TextSpan sourceSpan)
+        {
+            return new Span(sourceSpan.Start, sourceSpan.Length);
+        }
+
+        public string RelativePath { get; private set; }
+
+        public int LineNumber { get; private set; }
+
+        public int CharacterNumber { get; private set; }
+
+        public IPersistentSpan PersistentSpan { get; private set; }
+
+        public ClassifiedTextElement ClassifiedDefinition { get; private set; }
+
+        public Span ContextHighlightSpan { get; private set; }
+
+        public string DocumentGuid { get; private set; }
```

Comment:
Asked earlier, but what's the meaning of this DocumentGuid?

Reply:
(this is news to me that VS has a concept of a document GUID at all, so quite surprised.)